arviz/plots/backends/bokeh/pairplot.py

```diff
@@ -42,156 +49,123 @@ def plot_pair(
 
+    tmp_flat_var_names = None
+    if len(flat_var_names) == len(list(set(flat_var_names))):
+        source_dict = dict(zip(flat_var_names, [list(post) for post in infdata_group]))
     else:
-        max_plots = (
-            numvars ** 2 if rcParams["plot.max_subplots"] is None else rcParams["plot.max_subplots"]
-        )
-        vars_to_plot = np.sum(np.arange(numvars).cumsum() < max_plots)
-        if vars_to_plot < numvars:
-            warnings.warn(
-                "rcParams['plot.max_subplots'] ({max_plots}) is smaller than the number "
-                "of resulting pair plots with these variables, generating only a "
-                "{side}x{side} grid".format(max_plots=max_plots, side=vars_to_plot),
-                UserWarning,
-            )
-            numvars = vars_to_plot
+        tmp_flat_var_names = ["{}__{}".format(name, str(uuid4())) for name in flat_var_names]
+        source_dict = dict(zip(tmp_flat_var_names, [list(post) for post in infdata_group]))
+    if divergences:
+        divergenve_name = "divergences_{}".format(str(uuid4()))
+        source_dict[divergenve_name] = np.array(diverging_mask).astype(bool).astype(int).astype(str)
 
-        (figsize, _, _, _, _, _) = _scale_fig_size(figsize, textsize, numvars - 2, numvars - 2)
+    source = ColumnDataSource(data=source_dict)
 
-        if ax is None:
-            ax = []
-            for row in range(numvars - 1):
-                row_ax = []
-                for col in range(numvars - 1):
-                    if row < col:
-                        row_ax.append(None)
-                    else:
-                        ax_ = bkp.figure(
-                            width=int(figsize[0] / (numvars - 1) * dpi),
-                            height=int(figsize[1] / (numvars - 1) * dpi),
-                            **backend_kwargs
-                        )
-                        row_ax.append(ax_)
-                ax.append(row_ax)
-            ax = np.array(ax)
+    if divergences:
+        source_nondiv = CDSView(
+            source=source, filters=[GroupFilter(column_name=divergenve_name, group="0")]
+        )
+        source_div = CDSView(
+            source=source, filters=[GroupFilter(column_name=divergenve_name, group="1")]
+        )
 
-        tmp_flat_var_names = None
-        if len(flat_var_names) == len(list(set(flat_var_names))):
-            source_dict = dict(zip(flat_var_names, [list(post) for post in infdata_group]))
-        else:
-            tmp_flat_var_names = ["{}__{}".format(name, str(uuid4())) for name in flat_var_names]
-            source_dict = dict(zip(tmp_flat_var_names, [list(post) for post in infdata_group]))
-        if divergences:
-            divergenve_name = "divergences_{}".format(str(uuid4()))
-            source_dict[divergenve_name] = (
-                np.array(diverging_mask).astype(bool).astype(int).astype(str)
-            )
+    # pylint: disable=R1702
```

Comment:
can you use `# pylint: disable=too-many-nested-blocks`? this way we have an idea of what kind of check we are skipping.